cluster/partitioner/partitioner.go

```diff
@@ -4,49 +4,64 @@ import (
 	"fmt"
 
 	"github.com/Shopify/sarama"
+	"github.com/cespare/xxhash"
+	jump "github.com/dgryski/go-jump"
 	"github.com/raintank/schema"
 )
 
 type Partitioner interface {
-	Partition(schema.PartitionedMetric, int32) (int32, error)
+	Partition(schema.PartitionedMetric, int32) int32
 }
 
 type Kafka struct {
-	PartitionBy string
-	Partitioner sarama.Partitioner
+	PartitionBy     string
+	Partitioner     sarama.Partitioner
+	GetPartitionKey func(schema.PartitionedMetric, []byte) []byte
 }
 
 func NewKafka(partitionBy string) (*Kafka, error) {
+	kafka := Kafka{
+		PartitionBy: partitionBy,
```

Comment:
do we need to keep that string? i guess it doesn't matter much for the memory overhead, but it seems like after the partitioner is initialized this string is never used again.

Reply:
can be argued both ways. let's just leave as is